src/stactools/palsar/constants.py

```diff
@@ -1,27 +1,44 @@
+from datetime import datetime
+from typing import Optional
+
 from pystac import Link, Provider
 from pystac import ProviderRole as PR
 # from pystac.extensions.sar import
 from pystac.extensions.eo import Band
+from pystac.utils import str_to_datetime
+
+# TODO: Update if including PALSAR back to 2007
 
-ALOS_PALSAR_PLATFORM = "alos-2/alos"
-ALOS_PALSAR_INSTRUMENTS = ["PALSAR-2", "PALSAR"]
+ALOS_COLLECTION_START: Optional[datetime] = str_to_datetime(
```

Comment:
Should/can this be inferred?

Reply:
from @chuckwondo 

> I'd suggest tweaking your palsar constants a bit, both to improve consistency, and to avoid potential typing problems of your own:
```
ALOS_COLLECTION_START = str_to_datetime("2015-01-01T00:00:00Z")
ALOS_COLLECTION_END = str_to_datetime("2020-12-31T23:59:59Z")
ALOS_TEMPORAL_EXTENT: List[List[Optional[datetime]]] = [[
    ALOS_COLLECTION_START, ALOS_COLLECTION_END
]]
ALOS_SPATIAL_EXTENT = [[-180., 90., 180., -90.]]
```
> The tweaks above addresses the following 2 things:
First, your start and end dates are now definitely datetime values, not Optional[datetime] values. The issue with typing them as Optional is that anywhere you want to use them, and the target type is not Optional, you'll have to force type narrowing by using an if statement, like so:
```
if (ALOS_COLLECTION_START is not None):
    # ALOS_COLLECTION_START is now narrowed to `datetime`,
    # so mypy (or any type analyzer) knows it is not `None`
    # Without the `if`, the following won't type check if
    # the parameter of some function `f` is `datetime`
    # instead of `Optional[datetime]`:
    f(ALOS_COLLECTION_START)
```
> Second, ALOS_TEMPORAL_EXTENT is now consistent with the structure of ALOS_SPATIAL_EXTENT in that both are now nested lists. Alternatively, you could remove the outer list from both of them for consistency as well. Either way, making one non-nested and the other nested, while naming them both "extent"s may lead to confusion due to the inconsistency.
Then in src/stactools/palsar/stac.py, change this:
```
    extent = Extent(SpatialExtent(co.ALOS_SPATIAL_EXTENT),
                    TemporalExtent([co.ALOS_TEMPORAL_EXTENT]))
```
> to this (remove extraneous nesting around co.ALOS_TEMPORAL_EXTENT, since the constant now includes the additional nesting):
```
    extent = Extent(SpatialExtent(co.ALOS_SPATIAL_EXTENT),
                    TemporalExtent(co.ALOS_TEMPORAL_EXTENT))
```
> I recommend sticking with the nesting on the constants primarily because that is the "expectation" of an extent. While pystac allows removal of the nesting, that's only to accommodate "common mistakes" regarding the expectation of nesting, as evidenced by the code comments within pystac:
class SpatialExtent:
``` ...
    def __init__(
        self,
        bboxes: Union[List[List[float]], List[float]],
        extra_fields: Optional[Dict[str, Any]] = None,
    ) -> None:
        # A common mistake is to pass in a single bbox instead of a list of bboxes.
        # Account for this by transforming the input in that case.
        ...


class TemporalExtent:
    def __init__(
        self,
        intervals: Union[List[List[Optional[datetime]]], List[Optional[datetime]]],
        extra_fields: Optional[Dict[str, Any]] = None,
    ):
        # A common mistake is to pass in a single interval instead of a
        # list of intervals. Account for this by transforming the input
        # in that case.
        ...
```

Reply:
However since it currently works, I would like to merge as is, and target this for a refactor.